File: py/2609.py
def euclidean(n, m):
  q = n
  r = m

  while r > 0:
    temp = q
    q = r
    r = temp % r
  
  return q

def gcd(n, m):
  return euclidean(m, n) if n < m else euclidean(n, m)

def lcm(n, m):
  return (int)(n * m / gcd(n, m))

n, m = list(map(int, input().split()))

print(gcd(n, m))
print(lcm(n, m))

File: cpp/2609.cpp
#include <iostream>

using namespace std;

void input (int* i) {
  cin >> *i;
}

void print (int o) {
  cout << o << '\n';
}

int euclidean(int n, int m) {
  int q = n;
  int r = m;

  while(r > 0) {
    int temp = q;
    q = r;
    r = temp % r;
  }

  return q;
}

int gcd(int n, int m) {
  return n < m ? euclidean(m, n) : euclidean(n, m);
}

int lcm(int n, int m) {
  return (n * m) / gcd(n, m);
} 

int main() {
  int n, m;

  input(&n);
  input(&m);

  print(gcd(n, m));
  print(lcm(n, m));

  return 0;
}